nimble/host/src/ble_gattc.c

```diff
@@ -2453,7 +2453,8 @@ ble_gattc_disc_chr_uuid_rx_adata(struct ble_gattc_proc *proc,
     if (rc != 0) {
         /* Failure. */
         cbrc = ble_gattc_disc_chr_uuid_cb(proc, rc, 0, NULL);
-    } else if (ble_uuid_cmp(&chr.uuid.u, &proc->disc_chr_uuid.chr_uuid.u) == 0) {
+    } else if (ble_uuid_check(&chr.uuid.u) == 0 &&
+               ble_uuid_cmp(&chr.uuid.u, &proc->disc_chr_uuid.chr_uuid.u) == 0) {
```

Comment:
What does it mean 'malformed UUID' in this case? UUID received over the air is checked earlier so malformed UUID should be already rejected. One thing that `ble_uuid_init_from_buf` does not check is whether 16-bit UUID was not transferred as 128-bit UUID over the air which is incorrect and this is what should be fixed.